web/src/app/shared/configuration/field.service.ts

```diff
@@ -36,81 +36,115 @@ export interface IToolsEvent {
   conditions?: { advanced: boolean; search: string } | boolean;
 }
 
+const isVisibleField = (a: TFormOptions) => !a.ui_options?.invisible;
+const isAdvancedField = (a: TFormOptions) => isVisibleField(a) && a.ui_options?.advanced;
+const isHidden = (a: IFieldStack) => !!(a.ui_options?.invisible || a.ui_options?.advanced);
+
+const typeToControl: Partial<{ [key in TNForm | controlType]: controlType }> = {
+  bool: 'boolean',
+  int: 'textbox',
+  integer: 'textbox',
+  float: 'textbox',
+  string: 'textbox',
+  file: 'textarea',
+  text: 'textarea',
+};
+
+export const getControlType = (t: TNForm): controlType => typeToControl[t] || (t as controlType);
+
+const intPattern = () => new RegExp(/^[-]?\d+$/);
+const patternFn = {
+  integer: intPattern,
+  int: intPattern,
+  float: () => new RegExp(/^[-]?[0-9]+(\.[0-9]+)?$/),
+};
+
+export const getPattern = (t: TNForm): RegExp => (patternFn[t] ? patternFn[t]() : null);
+
+const fn = {
+  boolean: (v: boolean | null, d: boolean | null, r: boolean) => (String(v) === 'true' || String(v) === 'false' || String(v) === 'null' ? v : r ? d : null),
+  json: (v: string) => (v === null ? '' : JSON.stringify(v, undefined, 4)),
+  map: (v: object, d: object) => (!v ? d : v),
+  list: (v: string[], d: string[]) => (!v ? d : v),
+  structure: (v: any) => v,
+};
+
+export const getValue = (t: TNForm) => {
+  const def = (value: number | string) => (value === null || value === undefined ? '' : String(value));
+  return fn[t] ? fn[t] : def;
+};
+
+export const getKey = (n: string, sn: string) => (sn ? `${sn}/${n}` : n);
+
+export const getValidator = (required: boolean, min: number, max: number, type: TNForm) => ({
+  required,
+  min,
+  max,
+  pattern: getPattern(type),
+});
+
+const getField = (item: IFieldStack): IFieldOptions => ({
+  ...item,
+  key: getKey(item.name, item.subname),
+  value: getValue(item.type)(item.value, item.default, item.required),
+  validator: getValidator(item.required, item.limits?.min, item.limits?.max, item.type),
+  controlType: getControlType(item.type),
+  hidden: item.name === '__main_info' || isHidden(item),
+  compare: [],
+});
+
+const fo = (n: string) => (b: IFieldStack) => b.type !== 'group' && b.subname && b.name === n;
```

Comment:
I don't see the meaning of this function. It uses only one time

Reply:
just function as argument for filter by name for config field

Reply:
Just will name the function like "filterByName"